lightning/src/events/mod.rs

```diff
@@ -715,6 +717,29 @@ pub enum Event {
 		/// The `payment_id` to have been associated with payment for the requested invoice.
 		payment_id: PaymentId,
 	},
+	/// Indicates a [`Bolt12Invoice`] in response to an [`InvoiceRequest`] or a [`Refund`] was
+	/// received.
+	///
+	/// This event will only be generated if [`UserConfig::manually_handle_bolt12_invoices`] is set.
+	/// Use [`ChannelManager::send_payment_for_bolt12_invoice`] to pay the invoice or
+	/// [`ChannelManager::abandon_payment`] to abandon the associated payment.
+	///
```

Comment:
Oh, right. I forgot that that's checked at time of payment.

Reply:
Ended up reverting those changes and re-writing the `send_payment_for_bolt12_invoice` accordingly.

Reply:
Seems to be there: https://github.com/lightningdevkit/rust-lightning/pull/3078/commits/f2721aff76da8350109afc71ddcf3b81448ce92f

Reply:
Hmmm...`send_payment_for_bolt12_invoice` docs were changed to clarify when a `Bolt12PaymentError::UnexpectedInvoice` can occur, including the timeout scenarios. `StaleExpiration` is not explicitly mentioned because it is `pub(crate)`. I did remove the invoice-level expiration part since that's really a detail surfaced by `Event::PaymentFailed`, which is linked in the `send_payment_for_bolt12_invoice` docs.